blocks/library/embed/index.js

```diff
@@ -458,5 +470,11 @@ registerBlockType(
 	getEmbedBlockSettings( {
 		title: 'YouTube',
 		icon: 'video-alt3',
+		name: 'core-embed/youtube',
+		matchers: [
+			/^\s*(https?:\/\/(?:m\.|www\.)?youtube\.com\/watch\S+)\s*/i,
+			/^\s*(https?:\/\/(?:m\.|www\.)?youtube\.com\/playlist\S+)\s*/i,
+			/^\s*(https?:\/\/youtu\.be\/\S+)\s*/i,
```

Comment:
As of r41179 core is now supporting YouTube embed URL patterns too, eg: https://www.youtube.com/embed/ea2WoUtbzuw